File: auth-data-postgres-client/src/main/java/com/cartobucket/auth/postgres/client/repositories/ClientCodeRepository.java
package com.cartobucket.auth.postgres.client.repositories;

import com.cartobucket.auth.postgres.client.entities.ClientCode;
import io.quarkus.hibernate.orm.panache.PanacheRepositoryBase;
import jakarta.enterprise.context.ApplicationScoped;

import java.util.Optional;
import java.util.UUID;

@ApplicationScoped
public class ClientCodeRepository implements PanacheRepositoryBase<ClientCode, UUID> {
    public Optional<ClientCode> findByCode(String code) {
        return find("code = ?1", code).singleResultOptional();
    }

}

File: auth-data-postgres-client/src/main/kotlin/com/cartobucket/auth/postgres/client/repositories/ClientCodeRepository.kt
package com.cartobucket.auth.postgres.client.repositories

import com.cartobucket.auth.postgres.client.entities.ClientCode
import io.quarkus.hibernate.orm.panache.PanacheRepositoryBase
import jakarta.enterprise.context.ApplicationScoped
import java.util.Optional
import java.util.UUID

@ApplicationScoped
class ClientCodeRepository : PanacheRepositoryBase<ClientCode, UUID> {
    fun findByCode(code: String): Optional<ClientCode> = find("code = ?1", code).singleResultOptional()
}
